stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/Read.java

```diff
@@ -39,25 +42,56 @@
  */
 @BallerinaFunction(
         packageName = "ballerina.io",
-        functionName = "readBytes",
+        functionName = "read",
         receiver = @Receiver(type = TypeKind.STRUCT, structType = "ByteChannel", structPackage = "ballerina.io"),
-        args = {@Argument(name = "numberOfBytes", type = TypeKind.INT)},
+        args = {@Argument(name = "numberOfBytes", type = TypeKind.INT),
+                @Argument(name = "offset", type = TypeKind.INT)},
         returnType = {@ReturnType(type = TypeKind.BLOB),
-                @ReturnType(type = TypeKind.INT)},
+                @ReturnType(type = TypeKind.INT),
+                @ReturnType(type = TypeKind.STRUCT, structType = "IOError", structPackage = "ballerina.io")},
         isPublic = true
 )
-public class ReadBytes extends AbstractNativeFunction {
+public class Read extends AbstractNativeFunction {
 
     /**
      * Specifies the index which holds the number of bytes in ballerina.lo#readBytes.
      */
     private static final int NUMBER_OF_BYTES_INDEX = 0;
 
+    /**
+     * Specifies the offset of the array to read bytes.
+     */
+    private static final int OFFSET_INDEX = 1;
+
     /**
      * Specifies the index which contains the byte channel in ballerina.lo#readBytes.
      */
     private static final int BYTE_CHANNEL_INDEX = 0;
 
+    private static final Logger log = LoggerFactory.getLogger(Read.class);
+
+    /*
+     * Function which will be notified on the response obtained after the async operation.
+     *
+     * @param result context of the callback.
+     * @return Once the callback is processed we further return back the result.
+     */
+    /*private static EventResult readResponse(EventResult<Integer, EventContext> result) {
+        *//*
+         * The async response should go here
+         *//*
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        byte[] content = (byte[]) eventContext.getProperties().get(ReadBytesEvent.CONTENT_PROPERTY);
+        Integer numberOfBytes = result.getResponse();
+        return result;
+    }*/
```

Comment:
Shall we remove this commented code block?

Reply:
This needs to be uncommented upon the availability of the async functions